src/components/fields/DropdownCustom.js

```diff
@@ -0,0 +1,95 @@
+import React, {Component} from 'react';
+import PropTypes from 'prop-types';
+import {connectToContainer} from 'lib';
+import {MULTI_VALUED_PLACEHOLDER} from 'lib/constants';
+import Field from './Field';
+import DropdownWidget from '../widgets/Dropdown';
+import Text from './Text';
+
+class UnconnectedDropdownCustom extends Component {
+  constructor(props, context) {
+    super(props, context);
+
+    this.setValue = this.setValue.bind(this);
+    this.setLocals = this.setLocals.bind(this);
+
+    this.state = {custom: false};
+
+    this.setLocals(props);
+  }
+
+  componentWillReceiveProps(props) {
+    this.setLocals(props);
+  }
+
+  setLocals(props) {
+    this.value =
+      props.fullValue === undefined || // eslint-disable-line no-undefined
+      props.fullValue === MULTI_VALUED_PLACEHOLDER
+        ? this.props.defaultOpt
+        : props.fullValue;
+  }
+
+  setValue(value, custom = false) {
+    this.value = value;
+    this.setState({
+      custom: (custom || value === this.props.customOpt) && value !== '',
+    });
+    this.props.updateContainer({
+      [this.props.attr]: value === this.props.customOpt && !custom ? '' : value,
+    });
+  }
+
+  render() {
+    const {options, attr} = this.props;
+    const value =
+      (this.value === '' || !options.map(o => o.value).includes(this.value)) &&
+      this.state.custom
+        ? 'custom'
+        : this.value;
+
+    return (
+      <Field {...this.props}>
+        <DropdownWidget
+          backgroundDark={this.props.backgroundDark}
+          options={options}
+          value={value}
+          onChange={this.setValue}
+          clearable={this.props.clearable}
+          optionRenderer={this.props.optionRenderer}
+          valueRenderer={this.props.valueRenderer}
+          placeholder={this.props.placeholder}
+        />
+
+        {this.state.custom && (
+          <Text attr={attr} updatePlot={value => this.setValue(value, true)} />
```

Comment:
this doesn't seem to update as I type, only when I hit 'enter'

Reply:
Also, we could just have this send back `"custom"` when the field is empty, and preload it with `"custom"` so that the next field doesn't pop in and out of existence.